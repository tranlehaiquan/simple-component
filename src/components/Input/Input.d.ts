import * as React from 'react';

export interface InputProps {
  className?: string,
  placeholder?: string,
  readOnly?: boolean,
  name?: string,
  size?: 'md' | 'sm' | 'xs',
  disabled?: boolean,
  required?: boolean,
  maxLength?: number,
  minLength?: number,
  // clearable?: boolean,
  autoComplete?: boolean,
  nativeAutoComplete?: boolean,
  onFocus? (event: React.SyntheticEvent): void,
  onBlur? (event: React.SyntheticEvent): void,
  onChange? (event: React.SyntheticEvent): void,
  prefixIcon?: string | React.ReactNode,
  suffixIcon?: string | React.ReactNode,
  prefix?: string,
  suffix?: string
}

declare const Input: React.ComponentType<InputProps>;

export default Input;
