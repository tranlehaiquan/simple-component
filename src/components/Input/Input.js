import React, { forwardRef } from 'react';
import propTypes from 'prop-types';
import classnames from 'classnames';

const Input = forwardRef((props, ref) => {
  const { 
    placeholder, 
    readOnly, 
    name, 
    size, 
    disabled, 
    required, 
    className, 
    clearable,
    autoComplete,
    nativeAutoComplete,
    onBlur,
    onFocus,
    onChange,
    ...restProps
  } = props;
  const wrapperClassName = classnames('sp-input', {
    [`sp-input--size-${size}`]: size,
    ['sp-input--disabled']: disabled
  });

  function handleOnChange(e) {
    const { value } = e.target;

    onChange(value, e);
  }

  return (
    <div className={wrapperClassName}>
      <input
        className="sp-input__inner"
        placeholder={placeholder}
        readOnly={readOnly}
        name={name}
        disabled={disabled}
        required={required}
        ref={ref}
        onChange={handleOnChange}
        onBlur={onBlur}
        onFocus={onFocus}
        autoComplete={nativeAutoComplete.toString()}
        {...restProps}
      />
    </div>
  );
});

Input.propTypes = {
  placeholder: propTypes.string,
  readOnly: propTypes.bool,
  name: propTypes.string,
  size: propTypes.oneOf(['md', 'sm', 's']),
  disabled: propTypes.bool,
  required: propTypes.bool,
  maxLength: propTypes.number,
  minLength: propTypes.number,
  clearable: propTypes.bool,
  autoComplete: propTypes.bool,
  nativeAutoComplete: propTypes.bool,
  onFocus: propTypes.func,
  onBlur: propTypes.func,
  onChange: propTypes.func,
};

Input.defaultProps = {
  placeholder: '',
  readOnly: false,
  disabled: false,
  name: '',
  required: false,
  clearable: false,
  autoComplete: false,
  nativeAutoComplete: false,
  onFocus: () => undefined,
  onBlur: () => undefined,
  onChange: () => undefined,
};

export default Input;
