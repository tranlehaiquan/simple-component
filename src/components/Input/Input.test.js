import React from 'react';
import Input from './Input';
import renderer from 'react-test-renderer';

test('Render Input', () => {
	const component = renderer.create(
		<Input 
      placeholder="hello the world" 
      size="md"
      onChange={() => {}}
      onBlur={() => {}}
      onFocus={() => {}}
    />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
