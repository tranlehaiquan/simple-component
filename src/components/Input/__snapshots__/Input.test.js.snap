// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Input 1`] = `
<div
  className="sp-input sp-input--size-md"
>
  <input
    autoComplete="false"
    className="sp-input__inner"
    disabled={false}
    name=""
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="hello the world"
    readOnly={false}
    required={false}
  />
</div>
`;
